tensorflow/compiler/xla/service/gpu/cudnn_pad_for_convolutions.cc

```diff
@@ -211,29 +271,122 @@ static StatusOr<bool> PadForTensorCores(HloCustomCallInstruction* conv) {
     return false;
   }
 
-  // OK, let's do the transformation!
-  TF_RETURN_IF_ERROR(
-      PadConv(conv, new_lhs_shape, new_rhs_shape, new_result_shape));
+  new_input_shapes_ptr->push_back(new_lhs_shape);
+  new_input_shapes_ptr->push_back(new_rhs_shape);
   return true;
 }
 
-static std::vector<HloCustomCallInstruction*> GetRelevantConvs(
-    HloComputation* comp) {
-  std::vector<HloCustomCallInstruction*> convs;
-  for (HloInstruction* instr : comp->instructions()) {
-    if (IsCustomCallToDnnConvolution(*instr)) {
-      convs.push_back(Cast<HloCustomCallInstruction>(instr));
+// Adds padding to cudnn integer convolutions to make input and output feature
+// maps multiple of 4
+static StatusOr<bool> ResolvePadedShapesForIntegerConvolution(
+    HloCustomCallInstruction* conv, std::vector<Shape>* new_input_shapes_ptr,
+    Shape* new_result_shape_ptr) {
+  TF_ASSIGN_OR_RETURN(auto kind, GetCudnnConvKind(conv));
+  const Shape& input_shape = conv->operand(0)->shape();
+  const Shape& result_shape = conv->shape().tuple_shapes(0);
+
+  // Integer convolution only
+  if (!primitive_util::IsIntegralType(input_shape.element_type())) {
+    return false;
```

Comment:
This seems like an error. Maybe return a `InvalidArgument()` status?

Reply:
We only need to pad for integer convolutions.  Float convolutions don't need padding.  We just let them go.

Reply:
Yes, but the function name `ResolvePadedShapesForIntegerConvolution` indicates that it only handles integer convolutions, and it's the caller's responsibility to not pass in non-integer ones.

The current `return false;` behavior seems to better match a function name like `TryResolvePadedShapesForIntegerConvolution`.

Same for `ResolvePadedShapesForTensorCore`. Adding a `Try` is sufficient to clear the confusion.

Reply:
I agree.